ci/test-conformance-eks.sh

```diff
@@ -0,0 +1,186 @@
+    echo "CLUSTERNAME=${CLUSTER}" > ci_properties.txt
+
+    echo "=== Using the following awscli version ==="
+    aws --version
+
+    set +e
+    aws configure << EOF
+${AWS_ACCESS_KEY}
+${AWS_SECRET_KEY}
+${REGION}
+JSON
+EOF
+    set -e
+    printf "\n"
+    echo "=== Using the following kubectl ==="
+    which kubectl
+
+    echo '=== Creating a cluster in EKS ==='
+    eksctl create cluster \
+      --name ${CLUSTER} --region ${REGION} --version=${K8S_VERSION} \
+      --nodegroup-name workers --node-type ${AWS_NODE_TYPE} --nodes 2 \
+      --ssh-access --ssh-public-key ${SSH_PUBLIC_KEY} \
+      --managed
+    if [[ $? -ne 0 ]]; then
+        echo "=== Failed to deploy EKS cluster! ==="
+        exit 1
+    fi
+
+    mkdir -p ${WORKSPACE}/jenkins/out/
+    eksctl utils write-kubeconfig \
+      --cluster=${CLUSTER} --kubeconfig=${WORKSPACE}/jenkins/out/kubeconfig
+    export KUBECONFIG="${WORKSPACE}/jenkins/out/kubeconfig"
+
+    sleep 5
+    if [[ $(kubectl get svc) ]]; then
+        echo "=== EKS cluster setup succeeded ==="
+    else
+        echo "=== EKS kubectl is not configured correctly! ==="
+        exit 1
+    fi
+}
+
+function deliver_antrea_to_eks() {
+
+    echo "=== Configuring Antrea for cluster ==="
+
+    # EKS service CIDR is currently assigned by AWS,
+    # either from 172.20.0.0/16 or 10.100.0.0/16 depending on the range of VPC
+    # https://forums.aws.amazon.com/thread.jspa?messageID=859958
+    k8s_svc_addr=$(kubectl get svc -A | grep kubernetes | awk '{print $4}')
+
+    if [[ $k8s_svc_addr == 10.100.* ]]; then
+    	 k8s_svc_cidr='10.100.0.0/16'
+    elif [[ $k8s_svc_addr == 172.20.* ]]; then
+    	 k8s_svc_cidr='172.20.0.0/16'
+    else
+    	 echo "Cannot determine EKS serviceCIDR!"
+          exit 1
+    fi
+
+    set +e
+    worker_node_ip_1=$(kubectl get nodes -o wide | sed -n '2 p' | awk '{print $7}')
+    node_mtu=$(ssh -o StrictHostKeyChecking=no -l "ec2-user" $worker_node_ip_1 \
+      'export PATH=$PATH:/usr/sbin; ip a | grep -E eth0.*mtu | cut -d " " -f5')
+
+    sed -i "s|#defaultMTU: 1450|defaultMTU: ${node_mtu}|g" ../build/yamls/antrea-eks.yml
+    sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: ${k8s_svc_cidr}|g" ../build/yamls/antrea-eks.yml
+    echo "defaultMTU set as ${node_mtu}"
+    echo "seviceCIDR set as ${k8s_svc_cidr}"
+
+    kubectl apply -f ../build/yamls/antrea-eks.yml
+    set -e
+
+    kubectl rollout status --timeout=2m deployment.apps/antrea-controller -n kube-system
+    kubectl rollout status --timeout=2m daemonset/antrea-agent -n kube-system
+
+    echo "=== Antrea has been deployed for EKS cluster ${CLUSTER} ==="
+}
+
+function run_conformance() {
+    echo "=== Running Antrea Conformance and Network Policy Tests ==="
+
+    sudo ./${GIT_CHECKOUT_DIR}/ci/run-k8s-e2e-tests.sh \
+    --e2e-conformance --e2e-network-policy > eks-test.log
+
+    if grep -Fxq "Failed tests:" eks-test.log
+    then
+        echo "Failed cases exist."
+        touch TEST_FAILURE
```

Comment:
do you need to create a file here, or can you just use a variable?

Reply:
This was simply to conform with https://github.com/vmware-tanzu/antrea/blob/master/ci/jenkins/jobs/macros.yaml#L360 but yes I think a var would be sufficient